packages/matter.js/src/protocol/interaction/InteractionServer.ts

```diff
@@ -48,7 +50,34 @@ export const INTERACTION_PROTOCOL_ID = 0x0001;
 
 const logger = Logger.get("InteractionProtocol");
 
-export function ClusterServer<F extends BitSchema, SF extends TypeFromBitSchema<F>, A extends Attributes, C extends Commands, E extends Events>(
+function isConditionalAttribute(attribute: Attribute<any>) {
+    return (attribute as any).mandatoryIf !== undefined || (attribute as any).optionalIf !== undefined;
+}
+
+function isConditionalCommand(command: Command<any, any>) {
+    return (command as any).mandatoryIf !== undefined || (command as any).optionalIf !== undefined;
+}
+
+function isConditionalEvent(event: Event<any>) {
+    return (event as any).mandatoryIf !== undefined || (event as any).optionalIf !== undefined;
+}
+
+function validateConditionalEntity<F extends BitSchema, SF extends TypeFromBitSchema<F>>(
+    featureSets: ConditionalFeatureList<F>,
+    entityValue: any,
+    supportedFeatures: SF): boolean {
+    for (const features of featureSets) {
+        if (Object.keys(supportedFeatures).every(feature => !!features[feature] === !!supportedFeatures[feature])) {
```

Comment:
I was hoping if we used bit schema there's be proper set operations/testing.  Can this become supportedFeatures.hasAll(features)?

Reply:
supported Features in this place is "just a data object that has a defined set of keys with a defined valuetype". not more. so ... sorry but no ... We could do that but then on the Bitschea class whcih would be also a lot of overhead because then we always need to build the BitmapSchema instance ... would be bad performance wise. :-(

Reply:
Maybe a utility export then?  Would be nice to have this short and explicit, will probably need stuff like this for code gen too.  I can add later if you don't want to do now.